graph/graph_test.go

```diff
@@ -301,6 +301,6 @@ func tempGraph(t *testing.T) (*Graph, graphdriver.Driver) {
 }
 
 func nukeGraph(graph *Graph) {
```

Comment:
`nukeGraph` could certainly be renamed

As for `Root`, the approach I took in this refactor was to leave any variable or method exposed which is called by the push or pull code. This will make moving that code out easier. `Root` is used by the push flow here https://github.com/docker/docker/blob/master/graph/push.go#L251, although the best approach is probably to change that usage and hide the `Root` property.